monitors.go

```diff
@@ -171,11 +151,59 @@ func (client *Client) DeleteMonitor(id int) error {
 
 // GetMonitors returns a slice of all monitors
 func (client *Client) GetMonitors() ([]Monitor, error) {
+	return client.GetMonitorsWithOptions(MonitorQueryOpts{})
+}
+
+// MonitorQueryOpts contains the options supported by
+// https://docs.datadoghq.com/api/?lang=bash#get-all-monitor-details
+type MonitorQueryOpts struct {
+	GroupStates   []string
+	Name          string
+	Tags          []string
+	MonitorTags   []string
+	WithDowntimes *bool
+}
+
+// GetMonitorsWithOptions returns a slice of all monitors
+// It supports all the options for querying
+func (client *Client) GetMonitorsWithOptions(opts MonitorQueryOpts) ([]Monitor, error) {
 	var out reqMonitors
-	if err := client.doJsonRequest("GET", "/v1/monitor", nil, &out.Monitors); err != nil {
+	var query []string
+	if len(opts.Tags) > 0 {
+		value := fmt.Sprintf("tags=%v", strings.Join(opts.Tags, ","))
+		query = append(query, value)
+	}
+
+	if len(opts.GroupStates) > 0 {
+		value := fmt.Sprintf("group_states=%v", strings.Join(opts.GroupStates, ","))
+		query = append(query, value)
+	}
+
+	if len(opts.MonitorTags) > 0 {
+		value := fmt.Sprintf("monitor_tags=%v", strings.Join(opts.MonitorTags, ","))
+		query = append(query, value)
+	}
+
+	if opts.WithDowntimes != nil {
+		value := fmt.Sprintf("with_downtimes=%t", *opts.WithDowntimes)
+		query = append(query, value)
+	}
+
+	if opts.Name != "" {
```

Comment:
Works at the moment but needs to be changed either manually test if not nil if you refactor Name to be a pointer. Alternatively you can use the autogenereated getters and setters like so:

```go
if v, ok := opts.GetNameOk; ok {
    query = append(query, fmt.Sprintf("name=%v", strings.Join(v, ",")))
}
```